### DISCUSSION TITLE
VS Code integration doesn't work when using `-L`/`--lint`

### DISCUSSION THREAD

**Discussion by mayk0gan** [akaihola/darker#322] [Q&A] [answered]
Hi,

I'm trying to follow the instuctions in the readme file, but it doesn't seem to work.
My `settings.json`:

```
{
    "workbench.colorTheme": "Default Dark+",
    "files.trimTrailingWhitespace": true,
    "window.zoomLevel": 2,
    "editor.formatOnSave": true,
    "python.formatting.provider": "black",
    "python.formatting.blackPath": "/bin/darker",
    "python.formatting.blackArgs": [
        "--diff"
    ]
}
```

I've started with more complex configuration which didn't work too:
```
    "python.formatting.blackArgs": [
        "--rev",
        "dev",
        "--line-length=120",
        "--diff",
        "-L",
        "mypy",
        "-L",
        "flake8"
    ]
```

**Comment by mayk0gan** [comment]
from vscode's OUTPUT (python category):

```
formatting with black failed.
Error: Invalid patch mode '/' in: <PATH_TO_FILE>:35:1: E302 expected 2 blank lines, found 1
```

Where PATH_TO_FILE appears without '/' in in the beginning of the string.
Running the same command from the CLI (and from the same cwd) results more output

**Comment by mayk0gan** [comment]
Another update, it seems to be the `-L` flag to cause the mess (I tired with flake8 and mypy), without this flag it seems to work well..
Any ideas why?

**Comment by mayk0gan** [comment]
finally! you should add quotes around the linting tool/ its path.
For example:

`
"python.formatting.blackArgs": [
        "--rev",
        "develop",
        "-l",
        "120",
        "-L",
        "'flake8'"
]
`

But it seems that there isn't any integration between the chosen linting tool (flake8/ mypy) when running using darker, although they work great (and show suggestions under the PROBLEMS window) when run alone.

**Comment by akaihola** [comment, marked as answer]
Thanks a lot for your investigation @mayk0gan!

VSCode copies your `.py` file into a temporary `.tmp.py` file, submits that to Black (or in this case, Darker), and wants it to print on standard output the difference between the original and reformatted file – hence the `--diff` argument.

The problem with integrating other linting tools to the process is that their output is mixed in with the `--diff` output. So adding some `-L` options causes Darker to start acting very differently from Black, and VSCode doesn't understand that.

Maybe this is something we need to mention in the documentation.

On Linux, my VSCode configuration is:
```json
{
    // ...
    "python.formatting.blackArgs": [
        "--revision=master...",
        "--isort"
    ],
    "python.formatting.blackPath": "/home/akaihola/.virtualenvs/darker/bin/darker",
    "python.formatting.provider": "black",
    // ...
}
```
which seems to work all right.

**Comment by akaihola** [comment]
@mayk0gan since you're a VSCode and macos user, could you take a look at #67 as well to verify whether it's a valid issue report?

**Comment by mayk0gan** [comment]
I tried something similar to the configuration here (https://github.com/akaihola/darker/issues/67) and it works, I think that the main problem is with external linters (`-L` flag)

**Comment by akaihola** [comment]
Thanks @mayk0gan for that information! I'll close this issue and improve documentation in #254 instead.
